source/MissionAction.cpp

```diff
@@ -122,6 +123,12 @@ void MissionAction::Load(const DataNode &node, const string &missionName)
 			else
 				child.PrintTrace("Error: Unsupported use of \"system\" LocationFilter:");
 		}
+		else if(key == "teleport" && hasValue)
+		{
+			teleportPlanet = GameData::Planets().Get(child.Token(1));
+			if(child.Size() > 2 && child.Token(2) == "flagship only")
```

Comment:
I prefer to have this as a grandChild, not as a 2nd token.
And the keyword "planet" in the original request #6018 is still relevant if we want to support teleporting directly into a system as well in the future. (Even if this PR only adds the planet version.)

Reply:
Thinking of a syntax:
```
teleport
    planet planetname
    "flagship only"
```

Reply:
Is this what you want?

Reply:
Or:
`"planet teleport" planetname "flagship only"`

Reply:
I would go with what you suggested first.

While we're talking about syntax, I'd also like to suggest using `relocate` as the keyword instead of `teleport`.

Reply:
ok

Reply:
done

Reply:
Relocate is indeed better as keyword. And I like the proposal for the syntax.